src/strategies/Primitives/LimitSwapExactInput.ts

```diff
@@ -0,0 +1,115 @@
+import { TokenArgs, PrimitiveParamType, BigIntish } from '@brinkninja/types'
+import Token from '../Token'
+import InputTokenPrimitive from './InputTokenPrimitive'
+
+export type LimitSwapExactInputArgs = {
+  priceCurveAddress: string
+  priceCurveParams: string
+  signer: string
+  tokenIn: TokenArgs
+  tokenOut: TokenArgs
+  tokenInAmount: BigIntish,
+  fillStateParams: FillStateParamsJSON
+}
+
+export type FillStateParamsJSON = {
+  id: BigIntish
+  startX96: BigIntish
+  sign: boolean
+}
+
+export const LimitSwapExactInputFunctionParams: PrimitiveParamType[] = [
+  {
+    name: 'priceCurveAddress',
+    type: 'address',
+    signed: true
+  },
+  {
+    name: 'priceCurveParams',
+    type: 'bytes',
+    signed: true
+  },
+  {
+    name: 'owner',
+    type: 'address',
+    signed: true
+  },
+  {
+    name: 'tokenIn',
+    type: 'Token',
+    signed: true
+  },
+  {
+    name: 'tokenOut',
+    type: 'Token',
+    signed: true
+  },
+  {
+    name: 'tokenInAmount',
+    type: 'uint256',
+    signed: true
+  },
+  {
+    name: 'fillStateId',
+    type: 'uint64',
+    signed: true
+  },
+  {
+    name: 'fillStateStartX96',
+    type: 'uint128',
+    signed: true
+  },
+  {
+    name: 'fillStateSign',
+    type: 'boolean',
+    signed: true
+  },
+  {
+    name: 'data',
+    type: 'UnsignedMarketSwapData',
+    signed: false
+  }
+]
+
+export default class LimitSwapExactInput extends InputTokenPrimitive {
+  public constructor ({
+    priceCurveAddress,
+    priceCurveParams,
+    signer,
+    tokenIn,
+    tokenOut,
+    tokenInAmount,
+    fillStateParams
+  }: LimitSwapExactInputArgs) {
+    super({
+      functionName: 'limitSwapExactInput',
+      type: 'swap',
+      requiresUnsignedCall: true,
+      paramsJSON: {
+        priceCurve: {
+          address: priceCurveAddress,
+          params: priceCurveParams
+        },
+        signer,
+        tokenIn: (new Token(tokenIn)).toJSON(),
+        tokenOut: (new Token(tokenOut)).toJSON(),
+        tokenInAmount: tokenInAmount?.toString(),
+        fillStateParams
```

Comment:
this should be handled the same way Token is handled, because they're both structs. Need to create a `FillStateParams` class that follows the same pattern as the `Token` class (here https://github.com/brinktrade/brink-sdk/blob/ca025529f9e39f4ac37dec285f887c26bd41db51/src/strategies/Token.ts#L3)

We should probably have a struct interface that these implement with toJSON() and toStruct() fns. Cool if you want to add. otherwise just implement the same fns without an explicit interface

Reply:
basically you want to create fill state with args and return the json type which is more strict. the point of this is so we can take BigIntish as inputs and always output string values. Also can handle other similar input/output transformations. So ultimately this will be `fillStateParams: (new FillStateParams(fillStateParams)).toJSON()`